documentation/USAGE.rst

```diff
@@ -535,68 +535,223 @@ Further demonstrations of spatially varying boundary condition can be found
+------------------------
+
+Nothing different needs to be done when 
+`applying Robin boundary conditions`_ at internal faces.
+
+.. note::
+
+   While we believe these derivations are "correct", they often do not 
+   seem to produce the intuitive result. At this point, we think this has 
+   to do with the pathology of "internal" boundary conditions, but remain 
+   open to other explanations. :term:`FiPy` was designed with diffuse 
+   interface treatments (phase field and level set) in mind and, as such, 
+   internal "boundaries" do not come up in our own work and have not 
+   received much attention.
+
+.. note::
+
+  One must be careful to distinguish between constraining internal cell
+  values during the solve step and simply applying arbitrary constraints
+  to a ``CellVariable``. Applying a constraint,
 
->>> var.constrain(value, where=mask)
+  >>> var.constrain(value, where=mask)
 
-simply fixes the returned value of ``var`` at ``mask`` to be
-``value``. It does not have any effect on the implicit value of ``var`` at the
-``mask`` location during the linear solve so it is not a substitute
-for the source term machinations described above. Future releases of
-:term:`FiPy` may implicitly deal with this discrepancy, but the current
-release does not. A simple example can be used to demonstrate this::
+  simply fixes the returned value of ``var`` at ``mask`` to be
+  ``value``. It does not have any effect on the implicit value of ``var`` at the
+  ``mask`` location during the linear solve so it is not a substitute
+  for the source term machinations described above. Future releases of
+  :term:`FiPy` may implicitly deal with this discrepancy, but the current
+  release does not. A simple example can be used to demonstrate this::
 
->>> m = Grid1D(nx=2, dx=1.)
->>> var = CellVariable(mesh=m)
+  >>> m = Grid1D(nx=2, dx=1.)
+  >>> var = CellVariable(mesh=m)
 
-Apply a constraint to the faces for a right side boundary condition
-(which works).
+  Apply a constraint to the faces for a right side boundary condition
+  (which works).
 
->>> var.constrain(1., where=m.facesRight)
+  >>> var.constrain(1., where=m.facesRight)
 
-Create the equation with the source term constraint described above
+  Create the equation with the source term constraint described above
 
->>> mask = m.x < 1.
->>> largeValue = 1e+10
->>> value = 0.25
->>> eqn = DiffusionTerm() - ImplicitSourceTerm(largeValue * mask) + largeValue * mask * value
+  >>> mask = m.x < 1.
+  >>> largeValue = 1e+10
+  >>> value = 0.25
+  >>> eqn = DiffusionTerm() - ImplicitSourceTerm(largeValue * mask) + largeValue * mask * value
 
-and the expected value is obtained.
+  and the expected value is obtained.
 
->>> eqn.solve(var)
->>> print var
-[ 0.25  0.75]
+  >>> eqn.solve(var)
+  >>> print var
+  [ 0.25  0.75]
 
-However, if a constraint is used without the source term constraint an
-unexpected value is obtained
+  However, if a constraint is used without the source term constraint an
```

Comment:
This is a little confusing, maybe layout two code blocks with all the code for both versions, but with no text interspersed

Reply:
Not sure I agree. I'm a strong believer in literate programming, but I'll look at making sure the presentation is as clear as possible.

Reply:
@wd15 I've added some clarify text in f5273cc. Please check.